test/integration/crud/crud_api.test.ts

```diff
@@ -1183,19 +887,12 @@ describe('CRUD API', function () {
 
       ops.push({ insertOne: { _id: 0, a: i } });
 
-      client.connect(function (err, client) {
-        const db = client.db();
-        expect(err).to.not.exist;
-
-        db.collection('t20_1').bulkWrite(
-          ops,
-          { ordered: true, writeConcern: { w: 1 } },
-          function (err) {
-            test.ok(err !== null);
-            client.close(done);
-          }
-        );
-      });
+      const db = client.db();
+      const err = await db
+        .collection('t20_1')
+        .bulkWrite(ops, { ordered: true, writeConcern: { w: 1 } })
+        .catch(err => err);
+      expect(err).to.be.instanceOf(MongoBulkWriteError);
```

Comment:
The original assertion `test.ok(err !== null);` won't work well with promises, so I changed it to `instanceof`.